### ISSUE TITLE
method `io.netty.util.concurrent.DefaultPromise#cancel` violates contrast?

### ISSUE THREAD

**Issue by laosijikaichele** [netty/netty#7712]
### Expected behavior
According the contrast from method `java.util.concurrent.Future#cancel` :

> After this method returns, subsequent calls to isDone will always return true. 

the console should print `true`
### Actual behavior
console print `false`
### Steps to reproduce
please use following reproducer code
### Minimal yet complete reproducer code (or URL to code)
```
import io.netty.util.concurrent.GlobalEventExecutor;
import io.netty.util.concurrent.Promise;

public class DefaultPromiseIsDoneTest {

    private final Promise<?> defaultPromise = GlobalEventExecutor.INSTANCE.newPromise();

    public static void main(String args[]) {
        DefaultPromiseIsDoneTest main = new DefaultPromiseIsDoneTest();
        main.isDoneTest();
    }

    private void isDoneTest() {
        defaultPromise.setUncancellable();
        defaultPromise.cancel(false);
        boolean isDone = defaultPromise.isDone();
        System.out.println(isDone);
    }
}

```
### Netty version
4.1.21
### JVM version (e.g. `java -version`)
1.8
### OS version (e.g. `uname -a`)
MacOS

Does  Netty change the contrast inherited from `java.util.concurrent.Future#cancel`?
There are also a discussion about this contrast in stackoverflow:
https://stackoverflow.com/questions/30140463/future-cancel-method-documentation

**Comment by laosijikaichele** [comment]
friendly pinging...
@normanmaurer @trustin @Scottmitch

**Comment by normanmaurer** [comment]
@laosijikaichele yes I think you are right but there is not much we can do without breaking users until we do a new major release.  Let me add it to the todo list.

**Comment by normanmaurer** [comment]
https://github.com/netty/netty/issues/5430#issuecomment-365854227
